operator/internal/decommissioning/statefulset_decommissioner_test.go

```diff
@@ -0,0 +1,396 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package decommissioning_test
+
+import (
+	"context"
+	_ "embed"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/go-logr/logr/testr"
+	"github.com/redpanda-data/common-go/rpadmin"
+	"github.com/stretchr/testify/suite"
+	corev1 "k8s.io/api/core/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/redpanda-data/helm-charts/pkg/helm"
+	"github.com/redpanda-data/helm-charts/pkg/kube"
+	redpandav1alpha2 "github.com/redpanda-data/redpanda-operator/operator/api/redpanda/v1alpha2"
+	"github.com/redpanda-data/redpanda-operator/operator/internal/decommissioning"
+	"github.com/redpanda-data/redpanda-operator/operator/internal/testenv"
+	internalclient "github.com/redpanda-data/redpanda-operator/operator/pkg/client"
+	"github.com/redpanda-data/redpanda-operator/operator/pkg/functional"
+)
+
+//go:embed role.yaml
+var decommissionerRBAC []byte
```

Comment:
Can this be changed to reference `operator/config/rbac/decommissioner-role`?

Reply:
unfortunately not, from [godocs](https://pkg.go.dev/embed#:%7E:text=The%20patterns%20are%20interpreted%20relative%20to%20the%20package%20directory%20containing%20the%20source%20file.%20The%20path%20separator%20is%20a%20forward%20slash%2C%20even%20on%20Windows%20systems.%20Patterns%20may%20not%20contain%20%E2%80%98.%E2%80%99%20or%20%E2%80%98..%E2%80%99%20or%20empty%20path%20elements%2C%20nor%20may%20they%20begin%20or%20end%20with%20a%20slash.:~:text=The%20//go%3Aembed%20directive%20accepts,or%20back%2Dquoted%20string%20literals.) for `embed`:

> The //go:embed directive accepts multiple space-separated patterns for brevity, but it can also be repeated, to avoid very long lines when there are many patterns. The patterns are interpreted relative to the package directory containing the source file. The path separator is a forward slash, even on Windows systems. Patterns may not contain ‘.’ or ‘..’ or empty path elements, nor may they begin or end with a slash. To match everything in the current directory, use ‘*’ instead of ‘.’. To allow for naming files with spaces in their names, patterns can be written as Go double-quoted or back-quoted string literals.